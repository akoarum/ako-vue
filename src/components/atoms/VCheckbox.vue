<template>
  <label class="vCheckbox">
    <input
      v-model="model"
      v-bind="$attrs"
      :value="value"
      type="checkbox"
      class="vCheckbox__input"
    />
    <span class="vCheckbox__label"><slot /></span>
  </label>
</template>

<script lang="ts">
import Vue, { PropType } from 'vue'

export default Vue.extend({
  inheritAttrs: false,
  model: {
    prop: 'checkedValues',
    event: 'change'
  },
  props: {
    value: { type: String, required: true },
    checkedValues: { type: Array as PropType<string[]>, required: true }
  },
  computed: {
    model: {
      get(): string[] {
        return this.checkedValues
      },
      set(newValue: boolean) {
        this.$emit('change', newValue)
      }
    }
  }
})
</script>

<style lang="scss" scoped>
.vCheckbox {
  position: relative;
  padding-left: 26px;
  font-size: $fontSize-pc;
}

.vCheckbox__input {
  position: absolute;
  left: 0;
  top: 0;
  width: 10px;
  height: 10px;
  opacity: 0;
}

.vCheckbox__input:checked + .vCheckbox__label::before {
  content: 'check_box';
  color: $color-theme;
}

.vCheckbox__label::before {
  position: absolute;
  left: 0;
  top: -0.03em;
  content: 'check_box_outline_blank';
  font-size: 2.4rem;
  font-weight: normal;
  line-height: 1;
  font-family: $fontFamily-icons;
  color: #888;
  transition: color 0.1s $easing;
}

[aria-invalid='true'] + .vCheckbox__label {
  color: $color-caution;
}

[aria-invalid='true'] + .vCheckbox__label::before {
  color: $color-caution;
}
</style>
