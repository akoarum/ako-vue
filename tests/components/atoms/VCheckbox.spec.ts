import { shallowMount } from '@vue/test-utils'
import VCheckbox from '~/components/atoms/VCheckbox.vue'

describe('VCheckbox', () => {
  let wrapper: any

  beforeEach(() => {
    wrapper = shallowMount(VCheckbox, {
      propsData: {
        value: 'test',
        checkedValues: []
      },
      slots: {
        default: 'テスト'
      }
    })
  })

  describe('initialized', () => {
    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })

    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
