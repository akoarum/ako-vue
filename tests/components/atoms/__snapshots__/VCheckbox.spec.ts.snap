// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VCheckbox initialized snapshot 1`] = `
<label
  class="vCheckbox"
>
  <input
    class="vCheckbox__input"
    type="checkbox"
    value="test"
  />
   
  <span
    class="vCheckbox__label"
  >
    テスト
  </span>
</label>
`;
